src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -0,0 +1,392 @@
+        /// </exception>
+        /// <exception cref="ArgumentNullException">
+        ///   Provided value is null.
+        /// </exception>
+        public void Add(string propertyName, string propertyValue) => _dictionary.Add(propertyName, new JsonString(propertyValue));
+
+        /// <summary>
+        ///   Adds the specified property as a <see cref="JsonBoolean"/> to the JSON object.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to add.</param>
+        /// <param name="propertyValue"><see cref="string"/> value of the property to add.</param>
+        /// <exception cref="ArgumentException">
+        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        public void Add(string propertyName, bool propertyValue) => _dictionary.Add(propertyName, new JsonBoolean(propertyValue));
+
+        /// <summary>
+        ///   Adds the specified property as a <see cref="JsonNumber"/> to the JSON object.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to add.</param>
+        /// <param name="propertyValue"><see cref="byte"/> value of the property to add.</param>
+        /// <exception cref="ArgumentException">
+        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        public void Add(string propertyName, byte propertyValue) => _dictionary.Add(propertyName, new JsonNumber(propertyValue));
+
+        /// <summary>
+        ///   Adds the specified property as a <see cref="JsonNumber"/> to the JSON object.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to add.</param>
+        /// <param name="propertyValue"><see cref="short"/> value of the property to add.</param>
+        /// <exception cref="ArgumentException">
+        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        public void Add(string propertyName, short propertyValue) => _dictionary.Add(propertyName, new JsonNumber(propertyValue));
+
+        /// <summary>
+        ///   Adds the specified property as a <see cref="JsonNumber"/> to the JSON object.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to add.</param>
+        /// <param name="propertyValue"><see cref="int"/> value of the property to add.</param>
+        /// <exception cref="ArgumentException">
+        ///   Property name to set already exists if handling duplicates is set to <see cref="DuplicatePropertyNameHandling.Error"/>.
+        /// </exception>
+        public void Add(string propertyName, int propertyValue) => _dictionary.Add(propertyName, new JsonNumber(propertyValue));
```

Comment:
Where's the `DuplicatePropertyNameHandling` enum-based logic that lets me replace/ignore?

Reply:
Yes, in all of those places shoud be `Add`, instead of `_dictionary.Add` (like in Add(string propertyName, string propertyValue)), thanks for noticing this.